knative-operator/pkg/controller/knativeeventing/knativeeventing_controller.go

```diff
@@ -109,3 +122,68 @@ func (r *ReconcileKnativeEventing) configure(instance *eventingv1alpha1.KnativeE
 	}
 	return nil
 }
+
+// set a finalizer to clean up hanging resources when instance is deleted
+func (r *ReconcileKnativeEventing) ensureFinalizers(instance *eventingv1alpha1.KnativeEventing) error {
+	for _, finalizer := range instance.GetFinalizers() {
+		if finalizer == finalizerName {
+			return nil
+		}
+	}
+	log.Info("Adding finalizer to KnativeEventing")
+	instance.SetFinalizers(append(instance.GetFinalizers(), finalizerName))
+	return r.client.Update(context.TODO(), instance)
+}
+
+// general clean-up, like deletion of hanging resources
+func (r *ReconcileKnativeEventing) delete(instance *eventingv1alpha1.KnativeEventing) error {
+	finalizers := sets.NewString(instance.GetFinalizers()...)
+
+	if !finalizers.Has(finalizerName) {
+		log.Info("Finalizer has already been removed from KnativeEventing, nothing to do")
+		return nil
+	}
+
+	log.Info("Running cleanup logic for KnativeEventing")
+	if err := r.deleteHangingResources(instance); err != nil {
```

Comment:
This deletes the deployment but the controller that watches this deployment, pingsource-controller, recreates it.
In this stage, that controller is not deleted yet.

Looking for help to solve this.

@markusthoemmes @jcrossley3